enterprise/internal/own/background/recent_contributors.go

```diff
@@ -73,3 +88,36 @@ func (r *recentContributorsIndexer) indexRepo(ctx context.Context, repoId api.Re
 	commitCounter.Add(float64(len(commitLog)))
 	return nil
 }
+
+func isSubRepoPermsRepo(ctx context.Context, repoID api.RepoID, cache rcache.Cache, checker authz.SubRepoPermissionChecker) (isSubRepoPermsRepo bool, err error) {
+	cacheKey := strconv.Itoa(int(repoID))
+	// Look for the repo in cache to see if we have seen it before instead of hitting the DB.
+	val, ok := cache.Get(cacheKey)
+	if ok {
+		var isSubRepoPermsRepo bool
+		if err := json.Unmarshal(val, &isSubRepoPermsRepo); err != nil {
+			return false, err
+		}
+		return isSubRepoPermsRepo, nil
```

Comment:
I think we can spare an extra `return` and indentation like this, wdyt?
```suggestion
		var isSubRepoPermsRepo bool
		err := json.Unmarshal(val, &isSubRepoPermsRepo)
		return isSubRepoPermsRepo, err
```